### PR TITLE
[server] /session routes

### PR DESCRIPTION
**Pull request by gr2m**
@christophwitzko could you review my work so far?

### CODE REVIEW

File: server/lib/routes.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        return reply(Boom.notFound())
+      }
+      if (authorization.substr(0, 7) !== 'Bearer ') {
+        return reply(Boom.notFound())
+      }
+
+      var bearerToken = req.headers.authorization.substr(7)
+      request.cookie('AuthSession=' + bearerToken)
+      request.get('/_session', function (error, respones, body) {
+        if (error) {
```
Comment (christophwitzko) [inline server/lib/routes.js]: I think we should check here the `respones.statusCode`.

**Comment by christophwitzko** [comment]
Awesome work @gr2m! :relaxed:
The test setup looks really good. :+1:

File: server/lib/routes.js (hunk 1 above)
Comment (gr2m) [inline server/lib/routes.js]: and pass it to `Boom.wrap(error, response.statusCode)`

Comment (gr2m) [inline server/lib/routes.js]: Okay, I think I got what you mean:

```js
if (error) {
  return reply(Boom.wrap(error))
}

if (response.statusCode >= 400) {
  return reply(Boom.create(response.statusCode, body))
}
```

see 3b7ef46

Comment (christophwitzko) [inline server/lib/routes.js]: :ok_hand:

**Comment by gr2m** [comment]
Okay, this is good for review. @christophwitzko @christophwitzko

File: bin/server [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  if (err) {
+    console.error(err)
+  }
+})
+
+// enable CORS
+server.ext('onPreResponse', corsHeaders)
+
+spawnPouchdbServer(function (error, pouch) {
+  if (error) throw error
```
Comment (Charlotteis) [inline bin/server]: I feel like there may have been a discussion on if statements previously in slack.
Would:

```javascript
if (error) {
    throw error
}
```

be more readable here?

Same for line 61 :)

File: demo/index.html [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<!DOCTYPE html>
+<html>
```
Comment (Charlotteis) [inline demo/index.html]: <html lang='en'> would be good here for accessibility :)

File: server/lib/routes/session.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+  return [{
+    method: 'GET',
+    path: '/session',
+    handler: function (_req, reply) {
+      var headers = _req.headers
+      var authorization = headers.authorization
+      if (!authorization) {
+        return reply(Boom.notFound())
+      }
```
Comment (Charlotteis) [inline server/lib/routes/session.js]: Maybe a space between the `var`'s and the first and second if statement? Just for readability.

File: server/lib/routes/session.js [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    }
+  }, {
+    method: 'DELETE',
+    path: '/session',
+    handler: function (_req, reply) {
+      var headers = _req.headers
+      var authorization = headers.authorization
+      var options = {
+        url: '/_session'
+      }
```
Comment (Charlotteis) [inline server/lib/routes/session.js]: Maybe a linebreak between these variables and the if statement, for readability?

File: tests/unit/server/session-test.js [hunk 6]
```diff
@@ -0,0 +1,141 @@
+var nock = require('nock')
+var test = require('tap').test
+
+var getServer = require('../utils/get-server')
+var couchdbErrorTests = require('../utils/couchdb-error-tests')
+
+getServer(function (error, server) {
+  if (error) return test.fail(error)
```
Comment (Charlotteis) [inline tests/unit/server/session-test.js]: I think an:

```
if (thing) {
    do the thing
}
```

is more readable.

**Comment by gr2m** [comment]
Args sorry @Charlotteis I went ahead and totally forgot about your great comments. I'll incorporate them now

**Comment by Charlotteis** [comment]
@gr2m possible to add the changes I made in https://github.com/hoodiehq/hapi-account/pull/11#discussion_r37851779 as well? :)

**Pull request merged**